### ISSUE TITLE
Package marked as unlicensed despite having a valid LICENSE file at its root.

### ISSUE THREAD

**Issue by tgrapperon** [SwiftPackageIndex/SwiftPackageIndex-Server#1123]
Hello,

I've recently submitted my package to SPI:
https://swiftpackageindex.com/tgrapperon/swift-composable-environment

It appears currently as having no license:
<img width="546" alt="SPI" src="https://user-images.githubusercontent.com/35562418/119228728-b400d880-baea-11eb-9571-a53751051b8c.png">
despite having a GitHub generated MIT LICENSE file since the creation of the repo:
https://github.com/tgrapperon/swift-composable-environment/blob/main/LICENSE

The current commit of SwiftPackageIndex-Server is 3a05cd63d8ea912d8ade1adbd0305795fa6a0425

Am I doing something wrong, is it expected and fixes itself after some time, or is this a parsing issue?

**Comment by tgrapperon** [comment]
I see that the GitHub `about` frame is not showing any license for my repo, while it does for other projects, and I guess that's the root of the issue.
I'll investigate, but it's clearly on GitHub's side, not SPI, so I'm closing the issue.

**Comment by Sherlouk** [comment]
Unfortunately we do just use GitHub's API for license detection, so a failure is almost definitely on their side.

One way of validating this is by using the GitHub GraphQL Explorer. I've setup this example query which you can run here: https://docs.github.com/en/graphql/overview/explorer

```graphql
query { 
  broken: repository(name:"swift-composable-environment", owner: "tgrapperon") {
    licenseInfo {
      key
    }
  }
  working: repository(name:"Alamofire", owner: "Alamofire") {
    licenseInfo {
      key
    }
  }
}
```

This gives an output like this:

```
{
  "data": {
    "broken": {
      "licenseInfo": null
    },
    "working": {
      "licenseInfo": {
        "key": "mit"
      }
    }
  }
}
```

This is showing how, for the repository you've raised, the licenseInfo is coming back as null - essentially meaning that GitHub doesn't know what license you have defined. I've used Alamofire as an example of a working repository (who also happen to be using MIT).

You may wish to consider trying out a different license format, potentially copying the MIT one that Alamofire uses and changing out the name/year/etc.

Having looked at the license docs [here](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/licensing-a-repository) it looks like GitHub uses the Ruby gem [Licensee](https://github.com/licensee/licensee) so might be worth raising a bug with them.

Hope this helps @tgrapperon.

**Comment by tgrapperon** [comment]
Thanks @Sherlouk ! I've closed the issue while you were typing this answer.

I'll try things and see how it goes.
I've used GitHub LICENSE option while creating the repo, without touching anything, so I though I was good to go.
I'll update this issue if I find what happened.

Thanks again for your response!

**Comment by tgrapperon** [comment]
That's it!

The package was clearly matched as MIT with Licensee, so the issue was on GitHub's side.
I've updated the LICENSE on my local clone (I've removed the trailing new line that GitHub's online editor always adds), and I've pushed the updated file. It immediately appeared with the MIT License, so I guess GitHub only tries to parse the license when the LICENSE file is modified, and for some reason failed the first time when it created it itself. I don't know. I don't think it was the trailing new line as Licensee had no issues matching it to MIT with it.

I have no worries with SPI catching up later today.
Thanks for your help!
